scripts/check_codex_quota_budget.py

```
@@ -0,0 +1,558 @@
+    reset_at: datetime | None
+    window_minutes: int | None
+
+
+@dataclass(frozen=True)
+class QuotaReadResult:
+    snapshot: QuotaSnapshot | None
+    errors: tuple[str, ...] = ()
+    warnings: tuple[str, ...] = ()
+
+
+def parse_timestamp(value: Any) -> datetime | None:
```

Ruff (0.15.15)

[warning] Too many return statements (7 > 6)

(PLR0911)